libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/MapboxCarMapSurfaceUtils.kt

```diff
@@ -0,0 +1,32 @@
+@file:OptIn(MapboxExperimental::class)
+
+package com.mapbox.androidauto.car
+
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+
+fun MapboxCarMapSurface.handleStyleOnAttached(
+    block: (Style) -> Unit
+): OnStyleLoadedListener {
+    getStyle { block(it) }
+
+    return OnStyleLoadedListener {
+        getStyle { block(it) }
+    }.also {
+        mapSurface.getMapboxMap().addOnStyleLoadedListener(it)
+    }
```

Comment:
If the style is not loaded at this moment, will we be notified twice when it is? If yes than we should probably do something like this:
```suggestion
    mapSurface.getMapboxMap().getStyle()?.let(block)

    return OnStyleLoadedListener {
        block(mapSurface.getMapboxMap().getStyle()!!)
    }.also {
        mapSurface.getMapboxMap().addOnStyleLoadedListener(it)
    }
```